sale_order_product_recommendation/wizards/sale_order_recommendation.py

```diff
@@ -80,6 +80,8 @@ def _prepare_recommendation_line_vals(self, group_line, so_line=False):
     @api.onchange("order_id", "months", "line_amount")
     def _generate_recommendations(self):
         """Generate lines according to context sale order."""
+        # Clean lines for newId environment
+        self.line_ids = False
```

Comment:
Why here instead of inside `if isinstance(self.id, models.NewId):`
Else condition assign self.line_ids